src/test/java/org/opensearch/security/privileges/RestLayerPrivilegesEvaluatorTest.java

```diff
@@ -0,0 +1,170 @@
+
+    @Mock
+    private ClusterService clusterService;
+    @Mock
+    private ThreadPool threadPool;
+    @Mock
+    private AtomicReference<NamedXContentRegistry> namedXContentRegistry;
+    @Mock
+    private ConfigModel configModel;
+    @Mock
+    private DynamicConfigModel dcm;
+    @Mock
+    private PrivilegesEvaluatorResponse presponse;
+    @Mock
+    private Logger log;
+
+    private RestLayerPrivilegesEvaluator privilegesEvaluator;
+
+    private static final User TEST_USER = new User("test_user");
+
+    @Before
+    public void setUp() throws InstantiationException, IllegalAccessException {
+        MockitoAnnotations.openMocks(this);
+
+        ThreadContext context = new ThreadContext(Settings.EMPTY);
+        when(threadPool.getThreadContext()).thenReturn(context);
+
+        privilegesEvaluator = new RestLayerPrivilegesEvaluator(
+            clusterService,
+            threadPool,
+            mock(AuditLog.class),
+            mock(ClusterInfoHolder.class),
+            namedXContentRegistry
+        );
+        privilegesEvaluator.onConfigModelChanged(configModel);
+        privilegesEvaluator.onDynamicConfigModelChanged(dcm);
+
+        when(log.isDebugEnabled()).thenReturn(false);
+    }
+
+    @Test
+    public void testEvaluate_Initialized_Success() {
+        String action = "action";
+        SecurityRoles securityRoles = mock(SecurityRoles.class);
+        when(configModel.getSecurityRoles()).thenReturn(securityRoles);
+        when(configModel.getSecurityRoles().filter(Collections.emptySet())).thenReturn(securityRoles);
+        when(securityRoles.impliesClusterPermissionPermission(action)).thenReturn(false);
+        when(securityRoles.impliesLegacyPermission(action)).thenReturn(false);
+
+        PrivilegesEvaluatorResponse response = privilegesEvaluator.evaluate(TEST_USER, Set.of(action));
+
+        assertNotNull(response);
+        assertFalse(response.isAllowed());
+        assertFalse(response.getMissingPrivileges().isEmpty());
+        assertTrue(response.getResolvedSecurityRoles().isEmpty());
+        verify(configModel, times(3)).getSecurityRoles();
+    }
+
+    @Test(expected = OpenSearchSecurityException.class)
+    public void testEvaluate_NotInitialized_ExceptionThrown() throws Exception {
+        String action = "action";
+        privilegesEvaluator.evaluate(TEST_USER, Set.of(action));
+    }
+
+    @Test
+    public void testMapRoles_ReturnsMappedRoles() {
+        Set<String> mappedRoles = Collections.singleton("role1");
+        when(configModel.mapSecurityRoles(any(), any())).thenReturn(mappedRoles);
+
+        Set<String> result = privilegesEvaluator.mapRoles(any(), any());
+
+        assertEquals(mappedRoles, result);
+        verify(configModel).mapSecurityRoles(any(), any());
+    }
+
+    @Test
+    public void testEvaluate_Successful_NewPermission() {
+        String action = "hw:greet";
+        SecurityRoles securityRoles = mock(SecurityRoles.class);
```

Comment:
Just for my knowledge, so the are the mapped used for the security privilege evaluation? Cuz I think you were using the securityRoles here.

Reply:
`mapRoles` internally maps roles from securityRoles via `this.configModel.mapSecurityRoles(user, caller);`
Check this out: https://github.com/opensearch-project/security/blob/8557fce67470dcb14bf9440530ea76284693a0bd/src/main/java/org/opensearch/security/privileges/RestLayerPrivilegesEvaluator.java#L128